lib/Echidna/Campaign.hs

```diff
@@ -134,17 +145,18 @@ callseq v w ql = replicateM ql (evalStateT genTxM w) >>= \is -> use hasLens >>=
 
 -- | Run a fuzzing campaign given an initial universe state and some tests. Return the 'Campaign' state once
 -- we can't solve or shrink anything.
-campaign :: ( MonadCatch m, MonadRandom m, MonadReader x m, Has GenConf x, Has TestConf x, Has CampaignConf x)
+campaign :: ( MonadCatch m, MonadReader x m, Has GenConf x, Has TestConf x, Has CampaignConf x)
          => StateT Campaign m a -- ^ Callback to run after each state update (for instrumentation)
          -> VM                  -- ^ Initial VM state
          -> World               -- ^ Initial world state
          -> [SolTest]           -- ^ Tests to evaluate
          -> m Campaign
-campaign u v w ts = view (hasLens . to knownCoverage) >>= \c ->
-  execStateT runCampaign (Campaign ((,Open (-1)) <$> ts) c) where
-    step        = runUpdate (updateTest v Nothing) >> u >> runCampaign
+campaign u v w ts = view (hasLens . to knownCoverage) >>= \c -> do
+  g <- view (hasLens . to (fromMaybe (mkStdGen 0) . seed))
```

Comment:
do we wanna always use the same random? shouldn't we consider `newStdGen` instead?

Reply:
this should never happen when running `echidna-test`. `getStdGen` is inserted into `EConfig` if a seed is not specified in the config.

Reply:
we're already in a `MonadThrow` though, there's no reason for this to be silent